server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesCreationProcessor.java

```diff
@@ -152,6 +153,16 @@ private void ensureValidMailboxName(MailboxCreateRequest mailboxRequest, Mailbox
         if (name.contains(String.valueOf(pathDelimiter))) {
             throw new MailboxNameException(String.format("The mailbox '%s' contains an illegal character: '%c'", name, pathDelimiter));
         }
+
+        if (nameMatchesSystemMailbox(name)) {
```

Comment:
Disagree... We should be unable to create it only if the mailbox already exist.

Please add a test Showing that : 
 - I can create a Spam mailbox by JMAP
 - I can not create a SPam mailbox a second time...

(Your if should become something like : 

```
if (nameMatchesSystemMailbox(name) 
   && systemMailboxProvider(Role.fromName(name)).isEmpty()) {
...
```